backend/modules/notifications/service.py

```
@@ -1 +1,336 @@
+        }
+    
+    def respond_to_invitation_by_token(
+        self,
+        token: str,
+        action: str
+    ) -> Dict[str, Any]:
+        """Respond to invitation by token (for non-authenticated users)."""
+        invitation = self.invitation_repo.get_invitation_by_token(token)
+        
+        if not invitation:
+            raise ValueError("Invitation not found or already responded to")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)